### PR TITLE
Move pair inversion to individual plugins and add logging

### PR DESCRIPTION
**Pull request by peachbits**

### CODE REVIEW

**Comment by peachbits** [comment]
Still need to add logging

File: src/coinMarketCap.ts [hunk 1]
```diff
@@ -61,61 +65,65 @@ const _fetchQuote = async (currency: string, date: string): Promise<string> => {
...
-  if (currencyB === 'USD') {
-    return {
-      rate: aToUsdRate,
-      needsWrite: true
-    }
-  }
-  const bToUsdRate = await _fetchQuote(currencyB, date)
-  if (bToUsdRate === '') {
+  // Check if both codes are crypto
+  if (
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/coinMarketCap.ts:105]: no

File: src/coinMarketCap.ts [hunk 2]
```diff
@@ -61,61 +65,65 @@ const _fetchQuote = async (currency: string, date: string): Promise<string> => {
       },
       json: true
     }
-    const url = `https://pro-api.coinmarketcap.com/v1/cryptocurrency/quotes/historical?symbol=${currency}&time_end=${date}&count=1`
+    const url = `https://pro-api.coinmarketcap.com/v1/cryptocurrency/quotes/historical?symbol=${cryptoCode}&time_end=${date}&count=1&convert=${fiatCode}`
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/coinMarketCap.ts:69]: ignore for now

File: src/index.ts [hunk 3]
```diff
@@ -142,6 +142,11 @@ router.get('/exchangeRate', async function(req, res) {
       .status(400)
       .json({ error: 'Future date received. Must send past date.' })
   }
+  const log = (...args): void => {
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/index.ts:145]: won't fix in this pr

File: src/index.ts [hunk 4]
```diff
@@ -151,19 +156,24 @@ router.get('/exchangeRate', async function(req, res) {
   }
   try {
     if (response == null) {
-      response = await getFromDb(currencyA, currencyB, dateNorm)
+      response = await getFromDb(currencyA, currencyB, dateNorm, log)
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/index.ts:159]: not in the scope of this PR

File: src/index.ts [hunk 5]
```diff
@@ -61,17 +61,27 @@ interface ExResponse {
...
+  currencyB: string,
   date: string
 ): Promise<ExchangeResponse> {
   try {
     const exchangeRate: nano.DocumentGetResponse & {
       [pair: string]: any
     } = await dbRates.get(date)
-    if (exchangeRate[currencyPair] == null) {
-      return
+    if (exchangeRate[`${currencyA}_${currencyB}`] == null) {
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/index.ts:73]: not doing

File: src/utils.ts [hunk 6]
```diff
@@ -67,4 +67,274 @@ async function postToSlack(date: string, text: string): Promise<void> {
   }
 }
 
-export { normalizeDate, validateObject, postToSlack }
+const coinMarketCapFiatMap = {
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/utils.ts:70]: ignoring for now

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**